### ISSUE TITLE
Bug: I2C bus initialization fail

### ISSUE THREAD

**Issue by acrandal** [acrandal/RevEng_PAJ7620#56]
The PAJ7620 doesn't work well if you're using an external power source from an Arduino board. When powering the Arduino via USB and then connecting the 7620 to the Arduino it seems to work just fine. When I started working with the PAJ7620 on an external 5v source, the I2C interface would not fire up properly.

I spent hours trying different power sources, voltage levels, timing, multiple arduino boards (uno, pro micro, nano, mega, raw 328p) and nothing was consistent. I eventually got it to be consistent by booting and then resetting the arduino with the reset pin the I2C bus would connect up properly. This was very ugly, but working.

**Comment by acrandal** [comment]
/* There's two register banks (0 & 1) to be selected between.
   * BANK0 is where most data collection operations happen, so it's default.
   * Selecting the bank is done here twice for a reason. When the 7620 turns
   *  on, the I2C bus is sleeping. When you first read/write to the bus
   *  the 7620 wakes up, but it sometimes misses that first message.
   * Running the 7620 on an arduino with the USB power, a single call here
   *  usually works, but as soon as you use an external power bus it might
   */

By adding a second call in a row to selectRegisterBank(BANK0); in RevEng_PAJ7620::begin() this behavior was fixed.

This double call to select the register bank is present in other drivers. Seeed's driver and other ones all do this.

**Comment by acrandal** [comment]
This initialization behavior is obliquely noted in the PixArt v0.8 version of the documentation on page 32 under progreamming sequence and initialization:

`Step 1：Power On, The VBus must be power on before VDD.
Step 2：Wait 700us for PAJ7620U2 to stabilize.
Step 3：Write slave ID or I2C read command to process I2
C wake-up.
It’s recommend to read Reg_0x00. It will return “0x20” when wake-up finish.
By the way, There is no-ack from PAJ7620U2 before wake-up finish.`

Basically, there's a timing issue where you have to write to the sensor, wait a bit, and then proceed normally. This is in addition to the 700us delay to ensure the power has fully energized the device.

So, by writing the select bank twice in a row, the first one wakes up the i2c bus, and the second one is more successful at actually handling communications.

Bugfix incoming for a 1.4.1 release.

**Comment by acrandal** [comment]
bugfix merged to master - closing report.
